Libs/vtkSegmentationCore/vtkSegmentation.cxx

```diff
@@ -36,21 +36,44 @@
 #include <vtkImageThreshold.h>
 #include <vtkMath.h>
 #include <vtkMatrix4x4.h>
+#include <vtkMinimalStandardRandomSequence.h>
 #include <vtkNew.h>
 #include <vtkObjectFactory.h>
 #include <vtkPointData.h>
 #include <vtkPolyData.h>
+#include <vtkSingleton.h>
 #include <vtkStringArray.h>
 #include <vtkTransform.h>
 #include <vtkTransformPolyDataFilter.h>
 #include <vtkVersion.h>
 
 // STD includes
 #include <algorithm>
+#include <chrono>
 #include <functional>
 #include <sstream>
 
+// GDCM includes
+#if defined(SEGMENTATION_UUID_SEGMENT_IDS) && SEGMENTATION_UUID_SEGMENT_IDS
```

Comment:
simplify

Reply:
Does this also need to take into account whether Slicer is built with DICOM support? The code currently has some compile logic using things such as:

`Slicer_BUILD_DICOM_SUPPORT`

`#ifdef VTKITK_BUILD_DICOM_SUPPORT`

Reply:
We did not want to reimplement an ISO standard UUID generator but did not want to add a new mandatory dependency either. That's why the CMake option was added.

When vtkSegmentationCore is used within Slicer then the UUID generation code in GDCM is should be available even if Slicer is built without DICOM support (because Slicer is always built with ITK and Slicer's ITK always includes GDCM), but it would be nice to test it. If it turns out that the GDCM UUID generator is only available when DICOM support is enabled then UUID generation should be disabled (and the plain random sequence generator will be used).

Reply:
Slicer_BUILD_DICOM_SUPPORT only seems to impact DCMTK and not GCDM, so it doesn't look like we need to test if Slicer is built with DICOM support.

Reply:
Good, thanks for checking!